RiotSwiftUI/Modules/Authentication/ChoosePassword/View/AuthenticationChoosePasswordScreen.swift

```diff
@@ -31,23 +33,90 @@ struct AuthenticationChoosePasswordScreen: View {
     // MARK: Views
     
     var body: some View {
-        GeometryReader { geometry in
-            VStack {
-                ScrollView {
-                    mainContent
-                        .readableFrame()
-                        .padding(.horizontal, 16)
-                }
+        ScrollView {
+            VStack(spacing: 0) {
+                header
+                    .padding(.top, OnboardingMetrics.topPaddingToNavigationBar)
+                    .padding(.bottom, 36)
+                form
             }
+            .readableFrame()
+            .padding(.horizontal, 16)
         }
         .background(theme.colors.background.ignoresSafeArea())
         .alert(item: $viewModel.alertInfo) { $0.alert }
         .accentColor(theme.colors.accent)
     }
-    
-    @ViewBuilder
-    var mainContent: some View {
-        AuthenticationChoosePasswordForm(viewModel: viewModel)
+
+    /// The title, message and icon at the top of the screen.
+    var header: some View {
+        VStack(spacing: 8) {
+            OnboardingIconImage(image: Asset.Images.authenticationPasswordIcon)
+                .padding(.bottom, 8)
+
+            Text(VectorL10n.authenticationChoosePasswordInputTitle)
+                .font(theme.fonts.title2B)
+                .multilineTextAlignment(.center)
+                .foregroundColor(theme.colors.primaryContent)
+                .accessibilityIdentifier("titleLabel")
+
+            Text(VectorL10n.authenticationChoosePasswordInputMessage)
+                .font(theme.fonts.body)
+                .multilineTextAlignment(.center)
+                .foregroundColor(theme.colors.secondaryContent)
+                .accessibilityIdentifier("messageLabel")
+        }
+    }
+
+    /// The text field and submit button where the user enters an email address.
+    var form: some View {
+        VStack(alignment: .leading, spacing: 12) {
+            if #available(iOS 15.0, *) {
+                textField
+                    .onSubmit(submit)
+            } else {
+                textField
+            }
```

Comment:
The `.onSubmit` isn't needed for `RoundedBorderTextField` as it has its own `onCommit`, so the availability check can be removed if you would like.